deploy/manifests/dev-v2/us-east-2/cluster/kube-system/aws-auth.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: aws-auth
+data:
+  mapRoles: |
+    - groups:
+      - system:bootstrappers
+      - system:nodes
+      rolearn: arn:aws:iam::407967248065:role/nodes.cluster-api-provider-aws.sigs.k8s.io
+      username: system:node:{{EC2PrivateDNSName}}
+    - groups:
+      - system:bootstrappers
+      - system:nodes
+      rolearn: arn:aws:iam::407967248065:role/-nodegroup-iam-service-role_dev-v2-default_dev-v2-pool-0
```

Comment:
Can you point me to the code that creates the node groups please? 

Judging by the naming I suspect some name config is not set which causes the role name to start with `-`. On a side note in case it's useful, I used AWS TF modules and [changed names](https://github.com/filecoin-project/storetheindex/blob/main/deploy/infrastructure/prod/us-east-2/eks.tf#L18-L21) as much as possible to generate simple human readable names for node group and their roles.

Reply:
@masih There was a flag that was missed in the cluster.yaml that should allow us to add a suffix to that var. clusterapi just does not handle not adding a suffix gracefully. I will fix that when I bring a new cluster up to finish the rest of the flux config.